### ISSUE TITLE
Display current configuration shortcut not working

### ISSUE THREAD

**Issue by dscrofts** [kovidgoyal/kitty#5051]
**Describe the bug**
Pressing the display current configuration shortcut (⌥ + ⌘ + , on macOS as per https://sw.kovidgoyal.net/kitty/conf/) is not working and produces the following error:

```
Key action failed

debug_config
invalid literal for int() with base 10: ''

Press Enter to close
```

**To Reproduce**
1. Press the above-mentioned key combination.

**Environment details**
```
macOS Big Sur Version 11.6.5
kitty 0.25.0
```

**Additional context**
I tried to reproduce the problem with `kitty --config NONE` and the same error above is displayed.

Please let me know if I can provide any further information.

**Comment by kovidgoyal** [comment]
Run kitty from another terminal and see if more details about the error
are printed and post those, otherwise wait for th enext release where
the error message will have the option to display a full traceback.

**Comment by dscrofts** [comment]
I tried running kitty from Terminal.app and also get the same error unfortunately. I will wait for the next release and report back when there is more information available.

**Comment by kovidgoyal** [comment]
Just FYI, kitty has nightly builds available: https://sw.kovidgoyal.net/kitty/binary/

**Comment by NMAC427** [comment]
I tried to debug this issue and I determined the problem to be in the [`macos_version`](https://github.com/kovidgoyal/kitty/blob/master/kitty/utils.py#L976) function. 
The call to `platform.mac_ver()` ([source](https://github.com/python/cpython/blob/main/Lib/platform.py#L431)) returns the following tuple `('', ('', '', ''), '')`. This only happens in the release binary because the `platform` module relies `plistlib` to parse the plist file that contains the macOS version. But this module can't be imported in "hardened programs" (?).

```
ImportError: dlopen(/Volumes/kitty-0.25.0/kitty.app/Contents/Resources/Python/lib/kitty-extensions/pyexpat.so, 2): Library not loaded: @executable_path/../Frameworks/libexpat.1.dylib
  Referenced from: /Volumes/kitty-0.25.0/kitty.app/Contents/Resources/Python/lib/kitty-extensions/pyexpat.so
  Reason: no suitable image found.  Did find:
	file system relative paths not allowed in hardened programs
```

I'm not too familiar with how the kitty build system works, but if this can't easily be fixed, I would suggest to use `platform.version()` as a workaround. This returns the Darwin kernel version (eg. instead of macOS 12.1 it would return Darwin 21.2.0). Slightly less elegant, but at least it doesn't break kitty.
